horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/AlertingServiceImpl.java

```diff
@@ -233,33 +230,15 @@ DISTINCT ON(variable_id) variable_id AS variable,
         System.setProperty("polyglot.engine.WarnInterpreterOnly", "false");
     }
 
-    @WithRoles(extras = Roles.HORREUM_SYSTEM)
-    @Transactional
-    public void onLabelsUpdated(Dataset.LabelsUpdatedEvent event) {
+    public void onLabelValuesCalculation(Dataset.LabelsUpdatedEvent event) {
         boolean sendNotifications;
-        DataPointDAO.delete("dataset.id", event.datasetId);
         DatasetDAO dataset = DatasetDAO.findById(event.datasetId);
         if (dataset == null) {
-            // The run is not committed yet?
-            // Retry `horreum.alerting.updateLabel.retries` times before logging a warning
-            retryCounterSet.putIfAbsent(event.datasetId, new AtomicInteger(0));
-            int retryCounter = retryCounterSet.get(event.datasetId).getAndIncrement();
-            if (retryCounter < labelCalcRetries) {
-                Log.infof("Retrying labels update for dataset %d, attempt %d/%d", event.datasetId, retryCounter,
-                        this.labelCalcRetries);
-                vertx.setTimer(1000, timerId -> messageBus.executeForTest(event.datasetId, () -> Util.withTx(tm, () -> {
-                    onLabelsUpdated(event);
-                    return null;
-                })));
-                return;
-            } else {
-                //we have retried `horreum.alerting.updateLabel.retries` number of times, log a warning and stop retrying
-                Log.warnf("Unsuccessfully retried updating labels %d times for dataset %d. Stopping", this.labelCalcRetries,
-                        event.datasetId);
-                retryCounterSet.remove(event.datasetId);
-                return;
-            }
```

Comment:
I have removed this "retry" mechanism because TBH I don't get which specific situation could trigger it, this logic is always getting called when a new Dataset is created, either async or not. If the dataset is not present can only happen if the Run is trashed before the event is getting processed.. so the retry will never be useful. Any other thoughts?